rules/instance-metadata-api-access/raw.rego

```diff
@@ -1,95 +1,32 @@
 package armo_builtins
-# Check for images from blacklisted repos
 
-metadata_azure(z) = http.send({
-	"url": "http://169.254.169.254/metadata/instance?api-version=2020-09-01",
-	"method": "get",
-	"headers": {"Metadata": "true"},
-	"raise_error": true,	
-})
 
-metadata_gcp(z) = http.send({
-	"url": "http://169.254.169.254/computeMetadata/v1/?alt=json&recursive=true",
-	"method": "get",
-	"headers": {"Metadata-Flavor": "Google"},
-	"raise_error": true,	
-})
+deny[msg] {
+	obj = input[_]
+	is_cloud_provider_info(obj)
 
-metadata_aws(z) = metadata_object { 
-	hostname := http.send({
-	"url": "http://169.254.169.254/latest/meta-data/local-hostname",
-	"method": "get",
-	"raise_error": true,	
-    })
-	metadata_object := {
-		"raw_body": hostname.raw_body,
-		"hostname" : hostname.raw_body,
-		"status_code" : hostname.status_code
-	}
-}
+	obj.data.providerMetaDataAPIAccess == true
 
-azure_metadata[msga] {	
-	metadata_object := metadata_azure("aaa")
-	metadata_object.status_code == 200
-	node_name := metadata_object.body.compute.name
-	nodes := input[_]
-	nodes.metadata.name == node_name
-	msga := {
-		"alertMessage": sprintf("Node '%s' has access to Instance Metadata Services of Azure.", [node_name]),
-		"alert": true,
-		"prevent": false,
-		"alertScore": 1,
-		"failedPaths": [],
-		"fixPaths": [],
-		"alertObject": {
-			"k8SApiObjects": [nodes],
-			"externalObjects": {
-				"azureMetadata" : [metadata_object.body]
-			}
-		}
-	}
-}
+	# filter out host-sensor data
+    obj_filtered := json.filter(obj, ["apiVersion", "kind", "metadata"])
```

Comment:
Why filter out data?

Reply:
Data only includes "providerMetaDataAPIAccess", why do we need it?

Reply:
Because this is exactly what failed. If the user will have anything to do with this related resource, is with this field

Reply:
done